### PR TITLE
Modal for creating workout sessions

### PR DESCRIPTION
**Pull request by bryanansong**
## Description
In this pull request I created the modal which allows users to create and manage workout sessions based on a given template. It provides functionality to add, update, and remove sets of exercises, and upon completion, sends the session data to my backend server to be stored. In the backend, I created a file to be my controller for my workoutSessions operations, in this case, it receives data from the request and first validates all the information from the frontend. After this, I create a workout session object. You will notice that I have a mapping over setRecords ( later changed to workoutSets ). This is to first create each set before creating the workout session that has all the sets.
<what this pull request is doing>
<what will be done in later PRs and not included here>

## Test Plan

https://github.com/bryanansong/ZenFitness/assets/50881304/bcc9c4a3-07f2-4f26-8da4-60cb7b0f4a9a

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx [hunk 1]
```diff
@@ -107,6 +114,31 @@ const WorkoutTemplateModal: React.FC<WorkoutTemplateModalProps> = ({
           placeholder="Workout template title"
           className={styles.input}
         />
+        <div className={styles.radioSection}>
+          <label className={styles.label}>Make template Public ?</label>
+          <div className={styles.optionsContainer}>
+            <label className={styles.option}>
+              <input
+                type="radio"
+                value="true"
```
Comment (m5matthew) [inline frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx:123]: Is there a better way to do this instead of using two inputs?

Can you do `value={isPublic? 'true' : 'false'}`?

Comment (bryanansong) [reply frontend/src/components/WorkoutTemplateModal/WorkoutTemplateModal.tsx:123]: I don't think I understand the approach you are talking about @m5matthew
